docs/includes/apiargs-aggregate-option.yaml

```diff
@@ -0,0 +1,71 @@
+arg_name: option
+name: allowDiskUse
+type: boolean
+description: |
+  Enables writing to temporary files. When set to ``true``, aggregation stages
+  can write data to the ``_tmp`` sub-directory in the ``dbPath`` directory. The
+  default is ``false``.
+interface: phpmethod
+operation: ~
+optional: true
+---
+arg_name: option
+name: batchSize
+type: integer
+description: |
+  Specifies the initial batch size for the cursor. A batchSize of ``0`` means an
+  empty first batch and is useful for quickly returning a cursor or failure
+  message without doing significant server-side work.
+interface: phpmethod
+operation: ~
+optional: true
+---
+source:
+  file: apiargs-MongoDBCollection-common-option.yaml
+  ref: bypassDocumentValidation
+post: |
+  This only applies when using the :ref:`$out <agg-out>` stage.
+
+  Document validation requires MongoDB 3.2 or later: if you are using an earlier
+  version of MongoDB, this option will be ignored.
+---
+arg_name: option
+name: comment
+type: string
+description: |
+  Users can specify an arbitrary string to help trace the operation through the
+  database profiler, currentOp, and logs.
+interface: phpmethod
+operation: ~
+optional: true
+---
+arg_name: option
+name: explain
+type: boolean
+description: |
+  Specifies whether or not to return the information on the processing of the
+  pipeline.
+interface: phpmethod
+operation: ~
+optional: true
+---
+arg_name: option
+name: hint
+type: string|array|object
+description: |
+  The index to use. Specify either the index name as a string or the index key
+  pattern as a document. If specified, then the query system will only consider
+  plans using the hinted index.
+interface: phpmethod
+operation: ~
+optional: true
+---
+arg_name: option
+name: useCursor
```

Comment:
Per [`aggregate`](https://docs.mongodb.com/manual/reference/command/aggregate/#command-fields) command docs, non-cursor aggregations are no longer supported in 3.6+. That's the same version where collection-agnostic aggregations were introduced (e.g. [`$currentOp`](https://docs.mongodb.com/manual/reference/operator/aggregation/currentOp/)), so I can't think of a case where an application could legitimately use `useCursor: false` with a DB-level aggregation.

In 3.6+, the only time a cursor won't be returned is if `explain` is used, and that option is mutually exclusive with `useCursor`. IIRC, the operation class actually checks for that.

I realize `useCursor` is a common option on the `Aggregate` operation, but I think we should leave this on the collection method.

Reply:
Yes, that makes sense - I didn't realise the versions lined up like that. I moved the option back to the collection docs and added a note about servers > 3.6 throwing an error.